adr/2020_11_18_sending_emails_via_api.md

```diff
@@ -0,0 +1,53 @@
+# API - Sending emails
+
+* Status: proposed
+* Date: 2020-11-18
+
+## Context and Problem Statement
+
+To achieve 100% api coverage, we need to handle emails by api.
+
+## Considered Options
+
+### Using events
+
+Using events allow us to trigger any handler once operation is completed. 
+
+* Sylius\ApiBundle\Event\OrderCompleted
+
+Once this event is triggered we handle it with ```src/Sylius/Bundle/ApiBundle/EventHandler/OrderCompletedHandler.php```
+
+Then command ```src/Sylius/Bundle/ApiBundle/Command/SendOrderConfirmation.php ``` is dispatched, which triggers command handler (```src/Sylius/Bundle/ApiBundle/CommandHandler/SendOrderConfirmationHandler.php```)
+ 
+SendOrderConfirmationHandler is sending email based on its destination (in this case OrderConfirmation)
+
+```
+$this->emailSender->send(
+    Emails::ORDER_CONFIRMATION_RESENT,
+    [$order->getCustomer()->getEmail()],
+    [
+        'order' => $order,
+        'channel' => $order->getChannel(),
+        'localeCode' => $order->getLocaleCode(),
+    ]
+);
+```
+
+* Good, because its sending email based on triggered event, once understood it's simple to add new handlers.
```

Comment:
```suggestion
* Good, because it's easy to integrate other logic after the initial command will succeed.
* Good, because it's easy to trigger the same operation several times.
* Good, because it's easier to handle mailing server downs gracefully
```